### PR TITLE
DOCS: Trailing commas are no longer an issue

### PR DESCRIPTION
**Pull request by cafferata**
This pull request updates the documentation to reflect changes in v4.15.0, where `END` is no longer required in code examples, and trailing commas are no longer an issue. 🥳 

Fixes https://github.com/StackExchange/dnscontrol/issues/3245

### CODE REVIEW

File: documentation/code-tricks.md [hunk 1]
```diff
@@ -9,70 +9,6 @@ Solution: Use a "builder" to construct it for you.
 * [M365_BUILDER](language-reference/domain-modifiers/M365_BUILDER.md)
 * [SPF_BUILDER](language-reference/domain-modifiers/SPF_BUILDER.md)
 
-# Trailing commas
```
Comment (tlimoncelli) [inline documentation/code-tricks.md:12]: Problem: Someone sees an old dnsconfig.js with `END` and can't find it in the docs.

Suggestion:

# END

You may see `D()` statements written with `END` at the end.  For example:

{% code title="dnsconfig.js" %}
```javascript
D("example.com", REG_MY_PROVIDER, DnsProvider(DSP_MY_PROVIDER),
  A("foo", "1.2.3.4"),
END);
```
{% endcode %}

As of DNSControl v4.15.0 these `END` statements can be removed.  `END` is a no-op which was added for reasons that no longer matter.  They can be removed.

Comment (cafferata) [reply documentation/code-tricks.md:12]: Thank you, Tom, for your feedback! I’ve incorporated it into [this commit](https://github.com/StackExchange/dnscontrol/pull/3248/commits/14d7bd1f07833a2bf14ea866bf5745523bcc3c13). You can preview the updated documentation [here](https://docs.dnscontrol.org/~/revisions/EYzKhdIcHuRVpPYIhxuG/advanced-features/code-tricks#trailing-commas).

**Review by tlimoncelli — APPROVED** [review]

**Comment by jauderho** [comment]
So is it better to not have trailing commas going forward?

**Comment by tlimoncelli** [comment]
> So is it better to not have trailing commas going forward?

Both work.  I encourage people to include the trailing comma because it makes "git diff" easier to read.

Now that both work, there's no need for the "END" hack to force it to work.

**Comment by tlimoncelli** [comment]
Ready for merge?

**Comment by jauderho** [comment]
BTW, the last I checked, the output generated by `dnscontrol write-types` still outputs `END`.

**Comment by tlimoncelli** [comment]
> BTW, the last I checked, the output generated by dnscontrol write-types still outputs END.

Looks like it only outputs the description of the END command itself:

```
$ git co cafferata-docs/code-tricks-end
Already on 'cafferata-docs/code-tricks-end'
$ go install
$ dnscontrol write-types
Successfully wrote types-dnscontrol.d.ts
$ egrep END types-dnscontrol.d.ts  |grep -v D_EXTEND
 * `END` permits the last item to include a comma.
 * END)
declare const END: DomainModifier & RecordModifier;
```

That said, we can update the comment:

```
diff --git a/commands/types/others.d.ts b/commands/types/others.d.ts
index 185ede34..ee134c77 100644
--- a/commands/types/others.d.ts
+++ b/commands/types/others.d.ts
@@ -43,7 +43,9 @@ declare const CF_UNIVERSALSSL_ON: DomainModifier;
 declare function CLI_DEFAULTS(vars: Record<string, unknown>): void;
 
 /**
- * `END` permits the last item to include a comma.
+ * `END` permits the last item to include a comma.  As of v4.15.0 this
+ * is no longer needed as the JavaScript interpreter now accepts a
+ * trailing comma.
  *
  * ```js
  * D("foo.com", ...
```

**Comment by tlimoncelli** [comment]
Sorry to rush you but I'm doing a point release today.  I'm merging this.  If further improvements are needed, at least doc-related PRs hit the website without waiting for a release.

**Comment by cafferata** [comment]
> Ready for merge?

Yes, it's was ready for merge. 👍🏻

**Pull request merged**